TiFlash debug binary failed to run Tpc-H

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
# use debug mode tiflash
tiup playground nightly --tiflash.binpath ./tiflash/tiflash
tiup bench tpch prepare
tiup bench tpch run
mysql> alter database test set tiflash replica 1;
```

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

success

### 3. What did you see instead (Required)

```
[2025/01/23 15:06:52.076 +08:00] [ERROR] [BaseDaemon.cpp:366] [########################################] [source=BaseDaemon] [thread_id=396]
[2025/01/23 15:06:52.076 +08:00] [ERROR] [BaseDaemon.cpp:367] ["(from thread 230) Received signal Aborted(6)."] [source=BaseDaemon] [thread_id=396]
[2025/01/23 15:06:52.076 +08:00] [ERROR] [BaseDaemon.cpp:559] ["
  0x55e7b97a4a9a    faultSignalHandler(int, siginfo_t*, void*) [tiflash+216038042]
                    libs/libdaemon/src/BaseDaemon.cpp:210
  0x7faa7fedd6f0    <unknown symbol> [libc.so.6+255728]
  0x7faa7ff2a94c    __pthread_kill_implementation [libc.so.6+571724]
  0x7faa7fedd646    __GI_raise [libc.so.6+255558]
  0x7faa7fec77f3    abort [libc.so.6+165875]
  0x7faa7fec771b    __assert_fail_base.cold [libc.so.6+165659]
  0x7faa7fed6386    __assert_fail [libc.so.6+226182]
  0x55e7bbe90775    DB::HashPartitionWriter<std::__1::shared_ptr<DB::AsyncMPPTunnelSetWriter>>::partitionAndWriteBlocksV1() [tiflash+256849781]
                    dbms/src/Flash/Mpp/HashPartitionWriter.cpp:148
  0x55e7bbe90216    DB::HashPartitionWriter<std::__1::shared_ptr<DB::AsyncMPPTunnelSetWriter>>::flush() [tiflash+256848406]
                    dbms/src/Flash/Mpp/HashPartitionWriter.cpp:96
  0x55e7bbe90159    DB::HashPartitionWriter<std::__1::shared_ptr<DB::AsyncMPPTunnelSetWriter>>::write(DB::Block const&) [tiflash+256848217]
                    dbms/src/Flash/Mpp/HashPartitionWriter.cpp:140
  0x55e7bbe9aa79    DB::ExchangeSenderSinkOp::writeImpl(DB::Block&&) [tiflash+256891513]
                    dbms/src/Operators/ExchangeSenderSinkOp.cpp:37
  0x55e7bbbd65e2    DB::SinkOp::write(DB::Block&&) [tiflash+253990370]
                    dbms/src/Operators/Operator.cpp:199
  0x55e7bbbd3921    DB::PipelineExec::executeImpl() [tiflash+253978913]
                    dbms/src/Flash/Pipeline/Exec/PipelineExec.cpp:155
  0x55e7bbbd09a5    DB::PipelineExec::execute() [tiflash+253966757]
                    dbms/src/Flash/Pipeline/Exec/PipelineExec.cpp:115
  0x55e7bbbec34f    DB::PipelineTaskBase::runExecute() [tiflash+254079823]
                    dbms/src/Flash/Pipeline/Schedule/Tasks/Impls/PipelineTaskBase.h:76
  0x55e7bbbec19c    DB::PipelineTask::executeImpl() [tiflash+254079388]
                    dbms/src/Flash/Pipeline/Schedule/Tasks/Impls/PipelineTask.h:38
  0x55e7bbbfb2f6    DB::Task::execute() [tiflash+254141174]
                    dbms/src/Flash/Pipeline/Schedule/Tasks/Task.cpp:128
  0x55e7bbc06a4d    DB::CPUImpl::exec(std::__1::unique_ptr<DB::Task, std::__1::default_delete<DB::Task>>&) [tiflash+254188109]
                    dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPoolImpl.h:32
  0x55e7bbc06789    DB::TaskThreadPool<DB::CPUImpl>::handleTask(std::__1::unique_ptr<DB::Task, std::__1::default_delete<DB::Task>>&) [tiflash+254187401]
                    dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:105
  0x55e7bbc0636e    DB::TaskThreadPool<DB::CPUImpl>::doLoop(unsigned long) [tiflash+254186350]
                    dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:84
  0x55e7bbc05b91    DB::TaskThreadPool<DB::CPUImpl>::loop(unsigned long) [tiflash+254184337]
                    dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp:63
  0x55e7bbc08444    decltype(*std::declval<DB::TaskThreadPool<DB::CPUImpl>*>().*std::declval<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long)>()(std::declval<unsigned long>())) std::__1::__invoke[abi:ue170006]<void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long, void>(void (DB::TaskThreadPool<DB::CPUImpl>::*&&)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*&&, unsigned long&&) [tiflash+254194756]
                    /DATA/disk1/ra_common/tiflash-env-17/sysroot/bin/../include/c++/v1/__type_traits/invoke.h:308
  0x55e7bbc083bf    void std::__1::__thread_execute[abi:ue170006]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long, 2ul, 3ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long>&, std::__1::__tuple_indices<2ul, 3ul>) [tiflash+254194623]
                    /DATA/disk1/ra_common/tiflash-env-17/sysroot/bin/../include/c++/v1/__thread/thread.h:221
  0x55e7bbc08172    void* std::__1::__thread_proxy[abi:ue170006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (DB::TaskThreadPool<DB::CPUImpl>::*)(unsigned long), DB::TaskThreadPool<DB::CPUImpl>*, unsigned long>>(void*) [tiflash+254194034]
                    /DATA/disk1/ra_common/tiflash-env-17/sysroot/bin/../include/c++/v1/__thread/thread.h:232
  0x7faa7ff28c02    start_thread [libc.so.6+564226]"] [source=BaseDaemon] [thread_id=396]
```

### 4. What is your TiFlash version? (Required)

master

--- a/dbms/src/Flash/Mpp/HashPartitionWriter.cpp
+++ b/dbms/src/Flash/Mpp/HashPartitionWriter.cpp
@@ -127,14 +127,14 @@ WriteResult HashPartitionWriter<ExchangeWriterPtr>::write(const Block & block)
     if (rows > 0)
     {
         rows_in_blocks += rows;
-        if (data_codec_version == MPPDataPacketV1)
+        if (data_codec_version >= MPPDataPacketV1)
             mem_size_in_blocks += block.bytes();
         blocks.push_back(block);
     }
 
     auto row_count_exceed = static_cast<Int64>(rows_in_blocks) >= batch_send_min_limit;
     auto row_bytes_exceed
-        = data_codec_version == MPPDataPacketV1 && mem_size_in_blocks >= MAX_BATCH_SEND_MIN_LIMIT_MEM_SIZE;
+        = data_codec_version >= MPPDataPacketV1 && mem_size_in_blocks >= MAX_BATCH_SEND_MIN_LIMIT_MEM_SIZE;
 
     if (row_count_exceed || row_bytes_exceed)
         return flush();

